libbeat/processors/actions/changeFieldProcessor.go

```diff
@@ -0,0 +1,89 @@
+package actions
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/pkg/errors"
+)
+
+// fieldProccessorFunc defines how fields must be processed
+type fieldProccessorFunc func(event *beat.Event, field string) error
+
+type changeFieldProcessor struct {
+	Fields        []string
+	IgnoreMissing bool
+	FailOnError   bool
+	fieldProcess  string
+	changeFunc    fieldProccessorFunc
+}
+
+// NewChangeFieldProcessor is an umbrella method for processing events based on provided fields. Such as converting event keys to uppercase/lowercase
+func NewChangeFieldProcessor(c *conf.C, fieldProcess string, changeFunc fieldProccessorFunc) (processors.Processor, error) {
```

Comment:
Let's name it `alterFieldNameProcessor`, I don't think the current naming is self-explanatory.

The rest of the names should be change to be consistent with it.

`fieldProcess` should be `processorName`.

Reply:
I changed it to `NewAlterFieldProcessor`. The 'Name' seemed extra.

Reply:
Sure, even better if we extend its functionality in the future.